token/transport.go

```diff
@@ -102,11 +93,7 @@ func (vrb variableRequestBuilder) Build(original *http.Request, tr *Request) err
 		return nil
 	}
 
-	if vrb.required {
-		return xhttpserver.MissingVariableError{Variable: vrb.variable}
-	}
-
-	return nil
+	return xhttpserver.MissingVariableError{Variable: vrb.variable}
```

Comment:
This should just return nil?

AFAIU, variables (regardless of coming from header or parameter) should no longer be required.

If the logic is such that one of the two builders (header or query parameter) succeeds, then we should be good.

Reply:
This is a path variable, e.g. /foo/{bar}.  If it is missing, that's always a serverside error since it represents a misconfiguration.  It means that client code did not map the URI paths correctly.

Headers and parameters are separate, and are the subject of this PR.